tests/unit/test_schema_registry.py

```diff
@@ -0,0 +1,257 @@
+"""
+Copyright (c) 2024 Aiven Ltd
+See LICENSE for details
+"""
+from avro.compatibility import SchemaCompatibilityResult, SchemaCompatibilityType
+from karapace.compatibility import CompatibilityModes
+from karapace.compatibility.schema_compatibility import SchemaCompatibility
+from karapace.config import DEFAULTS
+from karapace.in_memory_database import InMemoryDatabase
+from karapace.schema_models import ParsedTypedSchema, SchemaVersion, TypedSchema, ValidatedTypedSchema
+from karapace.schema_registry import KarapaceSchemaRegistry
+from karapace.schema_type import SchemaType
+from karapace.typing import Subject, Version
+from unittest import mock
+from unittest.mock import MagicMock, Mock
+
+import logging
+
+LOG = logging.getLogger(__name__)
+
+
+class MockedSchemaRegistry:
```

Comment:
Can you extract an interface and try to use DI, not blocking on this since its matter of taste, but I find hard to understand what this object its trying to do and what is the expected behaviour

Reply:
Moved to `tests/integration/test_schema_compatibility.py`